{
  "url": "https://api.github.com/repos/quarkusio/quarkus/issues/38788",
  "repository_url": "https://api.github.com/repos/quarkusio/quarkus",
  "labels_url": "https://api.github.com/repos/quarkusio/quarkus/issues/38788/labels{/name}",
  "comments_url": "https://api.github.com/repos/quarkusio/quarkus/issues/38788/comments",
  "events_url": "https://api.github.com/repos/quarkusio/quarkus/issues/38788/events",
  "html_url": "https://github.com/quarkusio/quarkus/issues/38788",
  "id": 2135356425,
  "node_id": "I_kwDOCFbutM5_RvQJ",
  "number": 38788,
  "title": "Resteast Reactive: BlockingNotAllowedException on POST Method with Uni return type",
  "labels": [
    {
      "id": 985346626,
      "node_id": "MDU6TGFiZWw5ODUzNDY2MjY=",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/kind/question",
      "name": "kind/question",
      "color": "d876e3",
      "default": false,
      "description": "Further information is requested"
    },
    {
      "id": 5085228218,
      "node_id": "LA_kwDOCFbutM8AAAABLxpsug",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/triage/needs-feedback",
      "name": "triage/needs-feedback",
      "color": "FDE078",
      "default": false,
      "description": "We are waiting for feedback."
    }
  ],
  "state": "closed",
  "locked": false,
  "milestone": null,
  "comments": 3,
  "created_at": "2024-02-14T23:14:59Z",
  "updated_at": "2024-02-15T13:26:15Z",
  "closed_at": "2024-02-15T13:24:52Z",
  "active_lock_reason": null,
  "body": "### Describe the bug\r\n\r\nI'm encountering a BlockingNotAllowedException in my RESTEasy Reactive code while trying to return a Uni<String> from a POST endpoint with a request body. The endpoint is supposed to return a Uni created using Uni.createFrom().item(\"\"), yet I'm still facing this exception. The exception seem to be triggered when Vertx try to read the inputstream\r\n\r\n\r\n### Expected behavior\r\n\r\nNo exception or more specific for the actual problem\r\n\r\n### Actual behavior\r\n\r\nBlockingNotAllowedException: Attempting a blocking read on the io thread\r\n\r\n### How to Reproduce?\r\n\r\n\r\n```java\r\n@Path(\"/message\")\r\n@Produces({MediaType.APPLICATION_JSON})\r\n@Consumes({MediaType.APPLICATION_JSON})\r\npublic class Controller{\r\n\r\n      @POST\r\n      @Path(\"add\")\r\n      public Uni<String> test(String body)\r\n      {\r\n           return Uni.createFrom().item(\"res\");\r\n      }\r\n}\r\n```\r\n\r\nMake a request with postman.\r\n\r\nI'm pretty sure the problem don't come from the endpoint but maybe from the dependencies.\r\nI'm using quarkus-resteasy-reactive-jackson, quarkus-qpid-jms, quarkus-config-yaml, quarkus-websocket-client, lombok and mapstruct.\r\n\r\n\r\n### Output of `uname -a` or `ver`\r\n\r\n_No response_\r\n\r\n### Output of `java -version`\r\n\r\nopenjdk version 21.0.1\r\n\r\n### Quarkus version or git rev\r\n\r\nQuarkus 3.5.3\r\n\r\n### Build tool (ie. output of `mvnw --version` or `gradlew --version`)\r\n\r\nversion 3\r\n\r\n### Additional information\r\n\r\nIf i remove the body part of the endpoint the exception is not thrown.",
  "reactions": {
    "url": "https://api.github.com/repos/quarkusio/quarkus/issues/38788/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/quarkusio/quarkus/issues/38788/timeline",
  "performed_via_github_app": null,
  "state_reason": "completed"
}
